src/main/kotlin/parts/wisdom/arcraider/Mixes.kt

```diff
@@ -0,0 +1,61 @@
+package parts.wisdom.arcraider
+
+import me.joypri.Mix
+import me.joypri.Part
+import parts.wisdom.arcraider.visualization.Square
+import parts.wisdom.arcraider.visualization.Grid as VisualGrid
+import java.lang.IllegalArgumentException
+
+abstract class ArcMix(vararg parts: Part) : Mix(*parts) {
+
+    abstract operator fun invoke(grid: VisualGrid): VisualGrid
+}
+
+open class Grid(vararg parts: Part) : Mix(*parts) {
+    val width by Width
+    val height by Height
+}
+
+open class Coords(vararg parts: Part) : Mix(*parts) {
+    val x by X
+    val y by Y
+
+    operator fun component1() = x
+    operator fun component2() = y
+}
+
+open class Dot(vararg parts: Part) : ArcMix(*parts) {
+    val start by Start
+    val color by TheColor
+
+    override fun invoke(grid: VisualGrid): VisualGrid {
+        val newSquares = grid.squares.toMutableList()
+        newSquares.add(Square(start.x, start.y, color.ordinal))
+        return grid.copy(squares = newSquares)
+    }
+}
+
+open class Line(vararg parts: Part) : ArcMix(*parts) {
+    val start by Start
+    val color by TheColor
+    val direction by TheDirection
+    val length by Length
+
+    init {
+        if (length <= 0) throw IllegalArgumentException("Line must have a positive length!")
+    }
+
+    override fun invoke(grid: VisualGrid): VisualGrid {
+        val newSquares = grid.squares.toMutableList()
+        newSquares.add(Square(start.x, start.y, color.ordinal))
```

Comment:
I believe this is redundant with the `delta == 0` pass through the subsequent `for` loop.

Reply:
oh whoops this was leftover from copy paste :P